### DISCUSSION TITLE
Issue loading data generated by new version of PyG

### DISCUSSION THREAD

**Discussion by davidireland-iso** [pyg-team/pytorch_geometric#3517] [Q&A] [answered]
I have a data set of `Data` objects from the new version of PyTorch. I Pickled these on my workstation (which is using the latest version of PyG) and send them to my local machine. When I tried loading the data up using pickle on my local machine, which uses the previous version of PyG, I get the error `ModuleNotFoundError: No module named 'torch_geometric.data.storage`. I can't really update my PyG version on my local station as I have lots of these datasets generated using the previous version of PyG, so I am wondering whether there is a way around this so that I can load my new dataset in the old version of PyG? 

Thanks!

**Comment by rusty1s** [comment, marked as answer]
You can save and load them via:
```python
torch.save(data.to_dict(), path)
data = Data.from_dict(torch.load(path))
```

**Reply by davidireland-iso** [reply]
does this work if I have a list of data objects? should have clarified that in the original question, sorry. 

e: assuming not as I don't believe a list has a `.to_dict()` method. I am assuming I can loop through my list and replace each `Data` object with `.to_dict()` and then pickle that corresponding list?

**Reply by rusty1s** [reply]
Yes, as long as you store dictionaries instead of individual data objects. You can do the conversion to dictionaries inside a list comprehension.

**Reply by davidireland-iso** [reply]
yep, that worked! thanks a lot for the quick reply! :-)
